File: Array & Hashing/Contains Duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

// Time & Space complexity O(n)
 var containsDuplicate = function(nums) {
  const hashset = new Set();
  for (i = 0; i<nums.length; i++) {
      if (hashset.has(nums[i])) return true;
      hashset.add(nums[i]);
  }
  return false;
};

// Time complexity O(n*2) Space complexity O(1)

 var containsDuplicate = function(nums) {
  // const hashset = new Set();
  for (i = 0; i<nums.length; i++) {
      for (j=i+1; j<nums.length; j++) {
          if (nums[i] === nums[j]) {
              return true;
          }
      }
  }
  return false;
};

File: Array & Hashing/Contains Duplicate.py
# time complexity - o(n) | space complexity - o(n)
class Solution:
    def containsDuplicate(self, nums):
        hashset = set()
        for n in nums:
            if  n in hashset:
                return True
            hashset.add(n)
        return False

# result = Solution()
# a = result.containsDuplicate([1,3,4,1])
# print(a)


# result2 = Solution()
# b = result2.containsDuplicate([1,4,5,6,7,9])
# print(b)